pennylane/data/data_manager.py

```diff
@@ -96,17 +128,17 @@ def _refresh_data_struct():
 
 def _fetch_and_save(filename, dest_folder):
     """Download a single file from S3 and save it locally."""
-    response = requests.get(os.path.join(S3_URL, filename), timeout=5.0)
+    response = requests.get(os.path.join(S3_URL, quote(filename)), timeout=5.0)
```

Comment:
What does this do?

Reply:
certain characters like a plus sign (HeH+, for example) can't be used in URLs, so this reformats the URL before making the request to S3. This fixes the bug Soran hit while testing last week.